### PR TITLE
dump resources: remove calls to exit(1)

### PR DESCRIPTION
**Pull request by davidlunarg**

### CODE REVIEW

File: framework/decode/vulkan_replay_dump_resources.cpp [hunk 1]
```diff
@@ -1879,7 +1879,6 @@ VkResult VulkanReplayDumpResourcesBase::QueueSubmit(const std::vector<VkSubmitIn
                     GFXRECON_LOG_ERROR("Dumping draw calls failed (%s). Terminating.",
                                        util::ToString<VkResult>(res).c_str())
                     Release();
-                    exit(1);
```
Comment (bradgrantham-lunarg) [inline framework/decode/vulkan_replay_dump_resources.cpp:1882]: Removing these means that the message is no longer true.  But the message probably means that execution can't continue.  Doesn't this need to terminate somehow, either by `abort` or `throw` or some such?

Comment (bradgrantham-lunarg) [reply framework/decode/vulkan_replay_dump_resources.cpp:1882]: GFXRECON_LOG_FATAL?

**Comment by bradgrantham-lunarg** [comment]
Fixes #1761

File: framework/decode/vulkan_replay_dump_resources.cpp (hunk 1 above)
Comment (antonio-lunarg) [reply framework/decode/vulkan_replay_dump_resources.cpp:1882]: Looking at `framework/util/logging.h`, it seems there is no real difference between `GFXRECON_LOG_ERROR` and `GFXRECON_LOG_FATAL`. Do we want `FATAL` to terminate every time?

**Review by bradgrantham-lunarg — APPROVED** [review]

**Review by bradgrantham-lunarg — APPROVED** [review]

**Review by roberto-c — APPROVED** [review]

**Pull request merged**